apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/observer/TrimObserver.java

```diff
@@ -30,8 +30,8 @@
     private final TrimService trimService;
     private volatile boolean trimDerivedTables = true;
     private int trimFrequency;
-    private final Object lock = new Object();
-    private final Queue<Integer> trimHeights = new PriorityQueue<>(); // will sort heights from lowest to highest automatically
+//    private final Object lock = new Object();
+    private final Queue<Integer> trimHeights = new PriorityBlockingQueue<>(); // will sort heights from lowest to highest automatically
```

Comment:
Do not use blocking queue here, it will cause incorrect behavior during sharding